nlpserver.py

```diff
@@ -533,6 +548,425 @@ def fasttext():
+		objects = detector_results.detections
+	vector = embedding_result.embeddings[0].embedding
+	# print(embedding_result.embeddings[0].embedding.shape[0])
+	# Vector size for this layer (inumlayers - 1) is 1024
+	# This "should" return a Unit Vector so we can use "dot_product" in Solr
+	# in theory vision embedder here is already L2. But let's do it manually again.
+	normalized = preprocessing.normalize([vector], norm=params['norm'])
+	print(vector.shape[0])
+	# see https://nightlies.apache.org/solr/draft-guides/solr-reference-guide-antora/solr/10_0/query-guide/dense-vector-search.html
+	data['mobilenet']['vector'] = normalized[0].tolist()
+	data['mobilenet']['objects'] = objects
+	data['mobilenet']['modelinfo'] = {'version':app.config["MOBILENET_MODEL_NAME"]} 
+	data['message'] = 'done'
+	return jsonify(data)
+
+@app.route("/image/insightface", methods=['GET', 'POST'])
+def insightface():
+	# Import your Libraries 
+	import torch
+	from torchvision import transforms
+	from PIL import Image, ImageDraw
+	from pathlib import Path
+	import pandas as pd
+	import numpy as np
+	from pathlib import Path
+	import requests
+	import cv2
+	from io import BytesIO
+	from keras.preprocessing.image import img_to_array
+	import insightface
+	from insightface.app import FaceAnalysis
+	from sklearn import preprocessing
+	import json
+
+	intermediate_features = []
+
+	def loadImage(url, size = 640):
+		try:
+			response = requests.get(url)
+			response.raise_for_status()
+		except requests.exceptions.RequestException as err:
+			data['error'] =  err.strerror
+			return False
+	
+		img_bytes = BytesIO(response.content)
+		img = Image.open(img_bytes)
+		img = img.convert('RGB')
+		img = np.array(img)
+		img = img[:, :, ::-1].copy()
+		return img
+	
+	data = dict(default_data)
+	data['message'] = "Insightface -  Parameters: 'iiif_image_url'"
+	data['insightface'] = {}
+	data['insightface']['objects']  = []
+	data['insightface']['vector']  = []
+	data['insightface']['modelinfo']  = {}
+	params = {}
+	
+
+	if request.method == 'GET':
+		params['iiif_image_url'] = request.args.get('iiif_image_url')
+		params['norm'] = request.args.get('norm')
+	elif request.method == 'POST':
+		params = request.form # postdata
+	else:
+		data['error'] = 'Invalid request method'
+		return jsonify(data)
+
+	if not params:
+		data['error'] = 'Missing parameters'
+		return jsonify(data)
+
+	if not params['iiif_image_url']:
+		data['error'] = '[iiif_image_url] parameter not found'
+		return jsonify(data)
+
+	if not params['norm']:
+		params['norm'] = 'l2'
+
+	if params['norm'] and params['norm'] not in ['l1','l2','max']:
+		params['norm'] = 'l2'
+
+	
+	try:
+		# providers=['CUDAExecutionProvider', 'CPUExecutionProvider']
+		img = loadImage(params['iiif_image_url'], 640)
+		if img is not False:
+			app = FaceAnalysis(providers=['CPUExecutionProvider'])
+			# This will get all models. Inclussive age, gender. (bad juju) etc. But we won't return those
```

Comment:
Good that we're not returning these. It would be better if there was a way to prevent these factors from being processed at all, but I understand that might not be possible. Maybe a good reason to consider not using insightface at all?

Reply:
Hi, you are right but the reason I can not avoid it is the `Embedding` (for the vector/just image comparison), that one is not generated when I do not run Analysis.